### DISCUSSION TITLE
Multiple API calling same Cosmos Db instance

### DISCUSSION THREAD

**Discussion by nagg-hash** [Azure/azure-cosmos-dotnet-v3#3743] [General]
Currently I have a .net core API deployed and it is connected to a Cosmos DB (SQL API) using CosmosClient(Microsoft.Azure.Cosmos).
Lets say in this case API name as **API 1** and Cosmos DB name as **Cosmos 1**

Now there is a new requirement in some other API (lets say **API 2**) to add new endpoint which internally need data from the same Cosmos Db(**Cosmos 1**) for some internal data processing.

So what would be the better approach in terms of performance and standards to access **Cosmos 1** from **API 2**?
- Directly accessing Cosmos DB(**Cosmos 1**) from  **API 2** using CosmosClient(Microsoft.Azure.Cosmos)?
-  Calling **API1** endpoint over http to get necessary data?

**Comment by sourabh1007** [comment]
Sounds like application design question. It depends on you how you want to design your application. API2 should be dependent on API1 or not. Anyway, you are making CosmosDB call in both the cases.
